### PR TITLE
feat: import tokens new UI

### LINKED ISSUE #9097: Token Import UX for Mobile
### What is this about?

Similar to our token import UX on Extension, we should implement the same designs for Mobile , we want to allow users to import muti tokens.

### Scenario

_No response_

### Design

Design available here:
https://www.figma.com/file/TfVzSMJA8KwpWX8TTWQ2iO/Asset-list-and-details?type=design&mode=design&t=hSmF8ceCtF7iq2zm-0

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by salimtb**
## **Description**

this PR will enable the multi import tokens on mobile.
figma : https://www.figma.com/file/TfVzSMJA8KwpWX8TTWQ2iO/Asset-list-and-details?type=design&node-id=481-28961&mode=design

## **Related issues**

Fixes:  #9097 

## **Manual testing steps**

1. Go to add token page
2. search for tokens
3. add multiples one
4. Go to custom token tab
5. add token using address

### **Before**

https://github.com/MetaMask/metamask-mobile/assets/26223211/6a53681a-77c7-4a7a-8745-bce83625eacb

### **After**

https://github.com/MetaMask/metamask-mobile/assets/26223211/b7ff0960-95fa-441a-b62f-0ac56ef531cd

https://github.com/MetaMask/metamask-mobile/assets/26223211/115ade35-c619-408b-bb72-fad177146ed6

https://github.com/MetaMask/metamask-mobile/assets/26223211/85cee029-722a-4a4c-9527-c76e767bc933

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/UI/MultiAssetListItems/MultiAssetListItems.test.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+];
+
+const mockSelectedAsset = [
+  {
+    address: '0xdac17f958d2ee523a2206206994597c13d831ec7',
+    symbol: 'USDT',
+    name: 'Tether USD',
+    iconUrl: 'https://example.com/usdt.png',
+  },
+];
```
Comment (NicolasMassart) [inline app/components/UI/MultiAssetListItems/MultiAssetListItems.test.tsx:27]: C: This is a nice mock. Exactly what I was looking for in my previous comment. 👍

File: app/components/UI/MultiAssetListItems/MultiAssetListItems.test.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+jest.mock('../../../util/networks', () => ({
+  getNetworkImageSource: jest.fn().mockReturnValue('mockedImageSource'),
+}));
+
+describe('MultiAssetListItems', () => {
+  afterEach(() => {
+    (useSelector as jest.Mock).mockClear();
+  });
+
+  it('should render correctly', async () => {
```
Comment (NicolasMassart) [inline app/components/UI/MultiAssetListItems/MultiAssetListItems.test.tsx:49]:
```suggestion
  it('render matches previous snapshot', async () => {
```

File: app/components/UI/SearchTokenAutocomplete/index.tsx [hunk 3]
```diff
@@ -101,49 +121,95 @@ const SearchTokenAutocomplete = ({ navigation }: Props) => {
 
   const handleSelectAsset = useCallback(
     (asset) => {
-      setSelectedAsset(asset);
+      const assetAddressLower = asset.address.toLowerCase();
+
+      const newSelectedAsset = selectedAsset.reduce((acc, currentAsset) => {
```
Comment (NicolasMassart) [inline app/components/UI/SearchTokenAutocomplete/index.tsx:126]: S: use a more meaningful variable name than `acc`, a few more lettres are not more expensive ;)

File: e2e/specs/assets/import-tokens.spec.js
Comment (NicolasMassart) [inline e2e/specs/assets/import-tokens.spec.js:1]: Q: can we prevent using delays and await instead?
Delays are already in the code, I know, but they should not, I believe they are just hiding some wrong test practice. It's always source of flackiness and makes all the test last longer than necessary. We should avoid same as much as possible, even if it looks like a convenient way to fix some platform issues.

File: locales/languages/en.json [hunk 4]
```diff
@@ -351,12 +351,17 @@
...
+    "search_information_not_saved":"Your search information will not be saved.",
+    "import_token": "Would you like to import this token?",
     "tokens_detected_in_account": "{{tokenCount}} new {{tokensLabel}} found in this account",
     "token_toast": {
       "tokens_imported_title": "Imported Tokens",
       "tokens_imported_desc": "Successfully imported {{tokenSymbols}}",
       "token_imported_title": "Imported Token",
       "token_imported_desc": "Successfully imported {{tokenSymbol}}",
+      "token_imported_desc_1": "You’ve imported a token.",
+      "token_imported_desc_2": "You’ve imported {{tokensNumber}} tokens.",
```
Comment (NicolasMassart) [inline locales/languages/en.json:364]: C: ideally, the name of a string would tell what it's used for. Just adding a number doesn't provide this info.

**Review by NicolasMassart — COMMENTED** [review]
Nice work!
Some suggestions, questions and comments, but mainly for maintainability, [guidelines](https://github.com/MetaMask/contributor-docs/blob/main/docs/unit-testing.md#use-it-to-specify-the-desired-behavior-for-the-code-under-test) compliance and readability.
Well done!
Let me know when I can have a second review pass.

File: app/components/UI/AddCustomToken/index.js [hunk 5]
```diff
@@ -250,6 +316,10 @@ class AddCustomToken extends PureComponent {
     if (symbolWithoutSpaces.length === 0) {
       this.setState({ warningSymbol: strings('token.symbol_cant_be_empty') });
       validated = false;
+    } else if (symbol.length >= 11) {
```
[the remark this answers is not in the document]
Comment (salimtb) [reply app/components/UI/AddCustomToken/index.js:319]: this was a product decision , but i'll try to get more informations and get back to you

File: app/components/UI/AddCustomToken/index.js [hunk 6]
```diff
@@ -340,97 +420,146 @@ class AddCustomToken extends PureComponent {
     const colors = this.context.colors || mockTheme.colors;
     const styles = createStyles(colors);
 
+    const goToWebView = () => {
+      // TODO: This functionality exists in a bunch of other places. We need to unify this into a utils function
```
[the remark this answers is not in the document]
Comment (salimtb) [reply app/components/UI/AddCustomToken/index.js:424]: issue created [here](https://github.com/MetaMask/metamask-mobile/issues/9113)

File: app/components/UI/AddCustomToken/index.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (salimtb) [reply app/components/UI/AddCustomToken/index.js:319]: so regarding this one , it's a product decision

**Review by NicolasMassart — DISMISSED** [review]
Looks good to me.
Dedup required and then I guess I will have to review again (do not force push please)

**Review by NicolasMassart — DISMISSED** [review]
Looks good to me.

**Comment by chrisleewilcox** [comment]
- [ ] Tests on Android emulator
- [ ] Tests on all Networks in the Popular list that support import tokens
- [ ] Update automated test: this will likely break some of our automated tests

**Comment by salimtb** [comment]
hey @chrisleewilcox ,
you can find the result of the requested test [here](https://drive.google.com/drive/folders/10kAM77cScq005rEgJbvpcLDX8sfubR78?usp=drive_link)
note that I didn't record all the popular networks but I did the test 1 by 1

**Review by chrisleewilcox — DISMISSED** [review]
LGTM

**Review by chrisleewilcox — DISMISSED** [review]
LGTM

**Review by chrisleewilcox — APPROVED** [review]
LGTM

**Pull request merged**